File: HoursWorked.cpp
/*
									read file

Created by:		Mani Majd
Date:			August 8th 2017
Purpose:		Training

*/
#include<iostream>
#include<fstream> //ifstream : read the file     ofstream: write on the file
#include<String>
using namespace std;

//-------------------------------------- prototype --------------------------------------
void readFile();
//--------------------------------------------------------------------------------------

int main() {

	//ifstream input("hours.txt");
	readFile();

	system("PAUSE");
}//end of main

void readFile() {

	ifstream input("hours.txt");

	int employeeID, id;
	string name;
	double firstD, secondD, thirdD, fourthD, fifthD, totalHoures = 0.0;
	bool foundID = false;

	if (input.is_open()) {
		cout << "Enter an ID = ? " << endl;
		cin >> id;
		while (input >> employeeID >> name >>
			firstD >> secondD >> thirdD >> fourthD >> fifthD) {
			if (id == employeeID) {
				foundID = true;
				totalHoures = (firstD + secondD + thirdD + fourthD + fifthD);
				cout << name << " worked " << totalHoures << " hours ( " << totalHoures / 5 << " hours/day)" << endl;
			}//end of if ID block
		}// end of wile loop

		if (!foundID) {
			cout << "ID# " << id << " not found." << endl;
		}


		input.close();
	}//end of if open block
}

File: HoursWorked.java
/*
									read file

Created by:		Mani Majd
Date:			August 8th 2017
Purpose:		Training

*/

import java.io.*;
import java.util.*;
public class HoursWorked {

	public static void main (String[] args) {

		try {
			Scanner input = new Scanner (new File("hours.txt"));
			readFile(input);
		
		} catch (FileNotFoundException error) {

			System.out.println("could not find the file");
		}
		
	}//end of main


	public static void readFile(Scanner input){
		Scanner console = new Scanner(System.in);
		System.out.println("Enter an ID = ?");
		int employeeID = console.nextInt();
		boolean foundID = false; // Do we found the person?

		while(input.hasNextLine()){				// process for each employee
			
			// line = "123 Susan 12.5 8.17 7.6 3.2 0.0"
			String line = input.nextLine();
			Scanner lineScan = new Scanner(line);
			
			
			int id = lineScan.nextInt();
			if (id==employeeID){
				foundID = true;
				String name = lineScan.next();
				double totalHoures = 0.0;
				int day = 0;
				
				while (lineScan.hasNextDouble()){
					totalHoures += lineScan.nextDouble();
					day++;
				}
				
				System.out.printf("%s%s%.2f%s%.2f%s",name, " worked ", totalHoures, " hours (" , totalHoures/day, " hours/day)");
//				System.out.println(name+" worked "+ totalHoures+ " hours (" + totalHoures/day+ " hours/day)");
			}
			}// end of wile loop
		
		if(!foundID){
			System.out.println("ID# "+ employeeID + " not found.");
		}
		
		
		
	}
}
